### PR TITLE
feat: setting to disable automatic gcode loading on mobile

### PR DESCRIPTION
**Pull request by kieraneglin**
Works toward #934.

Automatic gcode loading works well on computers, but on mobile browsers it can take away control for several seconds, especially on remote connections. This PR adds a new setting to allow different loading behaviours on mobile vs. desktop.

Note: the label text could definitely use an editorial review.

Signed-off-by: Kieran Eglin <kieran.eglin@gmail.com>

Screen Shot 2023-02-04 at 9 36 00 AM
Screen Shot 2023-02-04 at 9 36 04 AM

### CODE REVIEW

File: src/components/settings/GcodePreviewSettings.vue [hunk 1]
```diff
@@ -270,6 +282,22 @@ export default class GcodePreviewSettings extends Vue {
       value,
       server: true
     })
+
+    if (!value) {
+      this.autoLoadMobileOnPrintStart = false
+    }
```
Comment (kieraneglin) [inline src/components/settings/GcodePreviewSettings.vue:288]: It makes sense to me that `autoLoadOnPrintStart` should override `autoLoadMobileOnPrintStart`, so I've added this to ensure that mobile loading is disabled when `autoLoadOnPrintStart` is disabled

File: src/components/settings/GcodePreviewSettings.vue [hunk 2]
```diff
@@ -131,6 +131,18 @@
 
       <v-divider />
 
+      <app-setting :title="$t('app.setting.label.auto_load_mobile_on_print_start')">
+        <v-switch
+          v-model="autoLoadMobileOnPrintStart"
+          hide-details
+          class="mb-5"
+          @click.native.stop
+          :disabled="!autoLoadOnPrintStart"
```
Comment (kieraneglin) [inline src/components/settings/GcodePreviewSettings.vue:140]: For viz. I've added a disabled state to the slider that's dependent on `autoLoadOnPrintStart` being enabled

Comment (pedrolamas) [reply src/components/settings/GcodePreviewSettings.vue:139]: In these cases, we prefer to just hide the control!

**Comment by pedrolamas** [comment]
Thank you @kieraneglin, can you please add a footer to the main PR comment with a "Signed-off-by" line, following the [CONTRIBUTING](https://github.com/fluidd-core/fluidd/blob/develop/CONTRIBUTING.md) guidelines (_you are missing the email address on yours!_)

Thank you!

File: src/components/settings/GcodePreviewSettings.vue (hunk 2 above)
Comment (kieraneglin) [reply src/components/settings/GcodePreviewSettings.vue:139]: Is that done with a simple v-if or is there some animation I have to consider?

**Comment by kieraneglin** [comment]
@pedrolamas added my email!

File: src/components/settings/GcodePreviewSettings.vue (hunk 2 above)
Comment (pedrolamas) [reply src/components/settings/GcodePreviewSettings.vue:139]: `v-if` it please!

File: src/components/settings/GcodePreviewSettings.vue [hunk 3]
```diff
@@ -131,6 +131,20 @@
...
+        <v-switch
+          v-model="autoLoadMobileOnPrintStart"
+          hide-details
+          class="mb-5"
+          @click.native.stop
+        />
+      </app-setting>
+
+      <v-divider />
+
```
Comment (kieraneglin) [inline src/components/settings/GcodePreviewSettings.vue:147]: @pedrolamas updated to use v-if!

**Review by matmen — DISMISSED** [review]
Nice, LGTM!

**Review by pedrolamas — APPROVED** [review]

**Pull request merged**